src/containers/Tenant/Query/SaveQuery/SaveQuery.tsx

```diff
@@ -0,0 +1,151 @@
+    const dispatch = useTypedDispatch();
+    const queryNameToEdit = useTypedSelector(selectQueryName);
+
+    const onSaveQueryClick = () => {
+        dispatch(setQueryAction('save'));
+        dispatch(clearQueryNameToEdit());
+    };
+
+    const onEditQueryClick = () => {
+        dispatch(saveQuery(queryNameToEdit));
+        dispatch(clearQueryNameToEdit());
+    };
+
+    const renderSaveButton = () => {
+        return (
+            <Button onClick={onSaveQueryClick} disabled={isSaveButtonDisabled}>
+                {i18n('action.save')}
+            </Button>
+        );
+    };
+
+    const renderSaveDropdownMenu = () => {
+        const items = [
+            {
+                action: onEditQueryClick,
+                text: i18n('action.edit-existing'),
+            },
+            {
+                action: onSaveQueryClick,
+                text: i18n('action.save-as-new'),
+            },
+        ];
+        return (
+            <DropdownMenu
+                items={items}
+                renderSwitcher={(props) => (
+                    <Button {...props} disabled={isSaveButtonDisabled}>
+                        {i18n('action.edit')}
+                    </Button>
+                )}
+                popupProps={{placement: 'top'}}
+            />
+        );
+    };
+
+    return queryNameToEdit ? renderSaveDropdownMenu() : renderSaveButton();
+}
+
+export function SaveQueryDialog() {
+    const savedQueries = useSavedQueries();
+    const dispatch = useTypedDispatch();
+    const singleClusterMode = useTypedSelector((state) => state.singleClusterMode);
+    const queryAction = useTypedSelector(selectQueryAction);
+    const [queryName, setQueryName] = React.useState('');
+    const [validationError, setValidationError] = React.useState<string | null>(null);
+
+    const validateQueryName = (value: string) => {
+        if (savedQueries.some((q) => q.name.toLowerCase() === value.trim().toLowerCase())) {
+            return i18n('error.name-exists');
+        }
+        return null;
+    };
+
+    const onCloseDialog = () => {
+        dispatch(setQueryAction('idle'));
+        setQueryName('');
+        setValidationError(null);
+    };
+
+    const handleQueryNameChange = (value: string) => {
+        setQueryName(value);
+        setValidationError(validateQueryName(value));
+    };
+
+    const onSaveClick = () => {
+        if (!queryName || validationError) {
+            return;
+        }
+
+        dispatch(saveQuery(queryName));
+        onCloseDialog();
+    };
+
+    return (
+        <Dialog
+            open={queryAction === 'save'}
+            hasCloseButton={false}
+            size="s"
+            onClose={onCloseDialog}
+            onEnterKeyDown={onSaveClick}
+        >
+            <Dialog.Header caption={i18n('action.save')} />
+            <Dialog.Body className={b('dialog-body')}>
+                {singleClusterMode && <div className={b('dialog-row')}>{i18n('description')}</div>}
```

Comment:
This string is no longer depends on cluster mode, we always save in `localStorage`